enterprise/models.py

```diff
@@ -813,6 +821,34 @@ def __repr__(self):
         """
         return self.__str__()
 
+    def save(self, *args, **kwargs):
+        """
+        When saving an EnterpriseCourseEnrollment, update the account-level consent if needed.
+        """
+        enterprise_customer = self.enterprise_customer_user.enterprise_customer
+
+        if enterprise_customer.enforce_data_sharing_consent == EnterpriseCustomer.EXTERNALLY_MANAGED:
+            # If the consent is externally managed, create a record indicating external consent.
+            account_consent_equivalent = UserDataSharingConsentAudit.EXTERNALLY_MANAGED
+        elif self.consent_granted:
+            # If explicit consent has been provided, create a record indicating that.
+            account_consent_equivalent = UserDataSharingConsentAudit.ENABLED
+        elif self.consent_granted is not None:
+            # If consent was declined, create a record indicating that.
+            account_consent_equivalent = UserDataSharingConsentAudit.DISABLED
+        else:
+            # If no explicit consent state was set, indicate that.
+            account_consent_equivalent = UserDataSharingConsentAudit.NOT_SET
+
+        # This is an awful hack. Once we migrate to the planned generic consent architecture, it can DIAF.
```

Comment:
We'll consolidate to a single consent model that can have different scopes stored natively.